File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.py
class Solution(object):
    def searchRange(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        n=len(nums)
        low=0
        high=n-1
        lower=-1
        higher=-1
        while low<=high:
            mid=(low+high)//2
            if target<=nums[mid]:
                if target==nums[mid]:
                    lower=mid
                high=mid-1
            else:
                if target==nums[mid]:
                    lower=mid
                low=mid+1
        
        n=len(nums)
        low=0
        high=n-1
        
        while low<=high:
            mid=(low+high)//2
            if target>=nums[mid]:
                if target==nums[mid]:
                    higher=max(mid,higher)
                low=mid+1
            else:
                if target==nums[mid]:
                    higher=max(mid,higher)
                high=mid-1
        return [lower,higher]

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
       int n=nums.size();
       int startind=-1,endind=-1;
       for(int i=0;i<n;i++){
        if(nums[i]==target){
            if(startind==-1)
             startind=i;
           endind=max(endind,i);
        }
       }
       vector<int> ans={startind,endind};
       return ans; 
    }
};